### PR TITLE
Add a feature for the security profile origin change in 8.3

### PR DESCRIPTION
**Pull request by thecoop**
The version field is still needed for system index mappings, that will go away separately

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticsearchmachine test this please

**Comment by thecoop** [comment]
@elasticsearchmachine test this please

**Review by ldematte — APPROVED** [review]
LGTM

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/SecurityFeatures.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.elasticsearch.features.FeatureSpecification;
+import org.elasticsearch.features.NodeFeature;
+import org.elasticsearch.xpack.security.support.SecuritySystemIndices;
+
+import java.util.Map;
+
+public class SecurityFeatures implements FeatureSpecification {
+    @Override
+    public Map<NodeFeature, Version> getHistoricalFeatures() {
+        return Map.of(SecuritySystemIndices.SECURITY_PROFILE_ORIGIN_FEATURE, Version.V_8_3_0);
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/SecurityFeatures.java:20]:
```suggestion
        return Map.of(SecuritySystemIndices.SECURITY_PROFILE_ORIGIN_FEATURE, SecuritySystemIndices.VERSION_SECURITY_PROFILE_ORIGIN);
```

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Comment by thecoop** [comment]
@elasticmachine update branch

**Pull request merged**